### ISSUE TITLE
Deeplink invocation with query params as Bundle

### ISSUE THREAD

**Issue by rachitmishra** [CleverTap/clevertap-android-sdk#20]
Hello, 

Currently the deep-links are directly invoked using `fireUrlThroughIntent()`. Is it possible to forward the query params of the deep link url as `Bundle`? 

https://github.com/CleverTap/clevertap-android-sdk/blob/6cc25bc4b6acb2d89e44c2a3450c6315f40c3f69/clevertap-android-sdk/src/main/java/com/clevertap/android/sdk/CTInAppBaseFragment.java#L93

Add?

```
private fun getIntentExtras(url: String): Bundle? {
    return try {
        val uri = Uri.parse(url)
        val queryKeys: Set<String> = uri.queryParameterNames
        Bundle().apply {
            queryKeys.forEach {
                putString(it, uri.getQueryParameter(it))
            }
        }
    } catch (e: NullPointerException || e: UnsupportedOperationException) {
        LogUtils.logException(e)
        null
    } 
}
```

**Comment by darshanclevertap** [comment]
Hi @rachitmishra ,

Could you also explain a use case where passing query params as Bundle would be more beneficial?

**Comment by rachitmishra** [comment]
Hi @darshanclevertap , 

So in case the action is set, the deep-linked activity is opened which is great.

Now if one needs the data for deep link url we need to implement `onNewIntent` in all the deeplink activities and reload the activity.

But usually deep-linked activities require some identifier or payload to load data. 

Ex. If any activity has to load a particular tag the deep-link URI can be specified as
myapp://mydeeplink?tag=1, and if we pass this while opening the activity using deeplink the activity can auto load the data.

**Comment by rachitmishra** [comment]
Another example can be opening a WebView activity with a deeplink which expects a URL in bundle.

Ex. deeplink: myapp://mywebviewactivity?url=google.com

Now with current implementation the WebView activity will open but it won't receive the URL and I have to add additional changes to `onNewIntent()` to handle the same.

**Comment by rachitmishra** [comment]
Hi @darshanclevertap any updates?

**Comment by darshanclevertap** [comment]
@rachitmishra Hi, we have taken this up for discussion with our product team. I will keep you updated if we add this to our enhancements list.

**Comment by darshanclevertap** [comment]
@rachitmishra Sorry for the delay, but after discussion with our product team, we will be taking this up as an enhancement and put it in our upcoming releases. I will be tagging this issue and will close it once the feature is live. Thank you for pointing this out to us! 😄

**Comment by rachitmishra** [comment]
@darshanclevertap That's a great news :) Thank you so much for picking this up. 🙌

**Comment by rachitmishra** [comment]
Hi, @darshanclevertap Apologies, but when are you planning to release this? 😅

**Comment by darshanclevertap** [comment]
@rachitmishra I'm really sorry, we already had locked our requirements for the release when this issue was released. This will be taken up in the next release for sure. Sorry for the delay and thank you for your patience.

**Comment by shah-nishant** [comment]
@darshanclevertap Any ETA for the next release? Some of requirements are dependent on this

**Comment by rachitmishra** [comment]
Hello @darshanclevertap We should be releasing this now. This is very small request and it has been more than a month :)

**Comment by darshanclevertap** [comment]
Hi @rachitmishra this will go with the next release of CleverTap Android SDK slated for release around 30th July 2019. Will keep you posted.

**Comment by shah-nishant** [comment]
Hi @darshanclevertap any updates on the release?

**Comment by darshanclevertap** [comment]
I've made the code changes, this release is a massive release with a big new feature. I apologize for the delay but we are still in the testing phase and are going to take another week for official release.

**Comment by darshanclevertap** [comment]
@rachitmishra @shah-nishant Thank you guys for your patience! This feature has been released in v3.6.0. Closing this.

**Comment by rachitmishra** [comment]
@darshanclevertap partieeeh!!! 🙌 🙌
